src/anet.c

```diff
@@ -381,19 +381,53 @@ static int anetSetReuseAddr(char *err, int fd) {
     return ANET_OK;
 }
 
-static int anetCreateSocket(char *err, int domain) {
+#define ANET_SOCKET_CLOEXEC 1
+#define ANET_SOCKET_NONBLOCK 2
+#define ANET_SOCKET_REUSEADDR 4
+static int anetCreateSocket(char *err, int domain, int type, int protocol, int flags) {
     int s;
-    if ((s = socket(domain, SOCK_STREAM, 0)) == -1) {
+
+    /* In general, SOCK_CLOEXEC won't have noticeable effect
+     * except for cases which really need this flag.
+     * Otherwise, it is just a flag that is nice to have.
+     * Its absence shouldn't affect a common socket's functionality.
+     */
+#ifdef SOCK_CLOEXEC
+    if (flags & ANET_SOCKET_CLOEXEC) {
+      type |= SOCK_CLOEXEC;
+      flags &= ~ANET_SOCKET_CLOEXEC;
+    }
+#endif
+
+#ifdef SOCK_NONBLOCK
+    if (flags & ANET_SOCKET_NONBLOCK) {
+      type |= SOCK_NONBLOCK;
+      flags &= ~ANET_SOCKET_NONBLOCK;
+    }
+#endif
+
+    if ((s = socket(domain, type, protocol)) == -1) {
         anetSetError(err, "creating socket: %s", strerror(errno));
         return ANET_ERR;
     }
 
+    if (flags & ANET_SOCKET_CLOEXEC && anetCloexec(s) == ANET_ERR) {
```

Comment:
Short-circuit `&&` doesn't seem to be existing style here, and IMO is less clear than
```
    if (flags & ANET_SOCKET_CLOEXEC) {
        if (anetCloexec(s) == ANET_ERR) {
...
```

Reply:
I think we like short-circuit style. It's in line with the style in this repo, which often prefers brevity. Double nested blocks are more complex IMO.

Reply:
OK - I was basing this comment on the (now-removed) code in this file - the `if (flags & ANET_CONNECT_NONBLOCK)` case.